app/bundles/LeadBundle/EventListener/SearchSubscriber.php

```diff
@@ -125,4 +139,256 @@ public function onBuildCommandList(MauticEvents\CommandListEvent $event)
             );
         }
     }
+
+    /**
+     * @param LeadBuildSearchEvent $event
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function onBuildSearchCommands(LeadBuildSearchEvent $event)
+    {
+        switch ($event->getCommand()) {
```

Comment:
This switch is really terrible. There are a parameters sometimes (`en_US`) which suggests it will not work in some cases - what about different languages?
Switch could be avoided with polymorphism or Template method - just create a configured class which will do its job.

Reply:
Translator defaults to local configured language so "other languages" is covered.  But we also want to ensure that english is covered to allow a "universal support" via API.

Reply:
I see. Current code structure does not allow me to refactor this. The code is in repository which could not be easily refactored to a service. So I fixed other issues.